test/e2e/image_pull_error_test.go

```diff
@@ -76,19 +78,18 @@ func TestImagePullError(t *testing.T) {
 	}
 
 	t.Log("When the images are not pulled, the revision should have error status.")
-	err = v1test.CheckRevisionState(clients.ServingClient, revisionName, func(r *v1.Revision) (bool, error) {
+	wantRevReasons := sets.NewString("ImagePullBackOff", "ErrImagePull")
+	if err := v1test.CheckRevisionState(clients.ServingClient, revisionName, func(r *v1.Revision) (bool, error) {
 		cond := r.Status.GetCondition(v1.RevisionConditionReady)
 		if cond != nil {
-			if cond.Reason == "ImagePullBackOff" || cond.Reason == "ErrImagePull" {
+			if wantRevReasons.Has(cond.Reason) {
 				return true, nil
 			}
-			return true, fmt.Errorf("the revision %s was not marked with expected error condition, but with (Reason=%q, Message=%q)",
-				revisionName, cond.Reason, cond.Message)
+			return true, fmt.Errorf("the Revision %s ReadyCondition = (Reason=%q, Message=%q), wantReasons: %v",
+				revisionName, cond.Reason, cond.Message, wantRevReasons.UnsortedList())
```

Comment:
unsorted will randomly order.  I'd based toward determinism in test code.

Reply:
it's one of, so :shrug: but sure